lms/djangoapps/badges/backends/tests/test_badgr_backend.py

```diff
@@ -199,3 +207,92 @@ def test_badge_creation_event(self, post):
                 'issuer': 'https://example.com/v1/issuer/issuers/test-issuer',
             }
         }, self.get_event())
+
+    @patch('requests.post')
```

Comment:
Rather than directly `@patch()`ing `requests.post`, these tests might turn out a little cleaner by employing [requests-mock](https://requests-mock.readthedocs.io/en/latest/), but it's not being used anywhere else in `edx-platform` so perhaps its use is being discouraged?

Probably want to wait for an upstream review before you change anything here. :)

Reply:
I think you can use `httpretty` as the existing edx-platform library for this unit test. See example at https://github.com/edx/edx-platform/blob/3152c14192dcccac79b7b7a76eba91b380eba78a/openedx/core/lib/tests/test_edx_api_utils.py#L23